dashboard/src/modules/components/TableComponent/Table.test.js

```diff
@@ -19,30 +14,65 @@ jest.mock("utils/api", () => {
   return {
     get: () => ({
       data: MOCK_DATA,
+      status: 200,
     }),
   };
 });
-
+test("Page heading is displayed on initial load", async () => {
+  render(<AppWrapper />);
+  await screen.findByText("pbench_user_benchmark1");
+  const heading = screen.getByRole("heading", { name: 'Results' });
+  expect(heading).toBeInTheDocument();
+});
 test("data from API is displayed on initial load", async () => {
   render(<AppWrapper />);
-  const benchmarkName = await screen.findByText("pbench_user_benchmark1");
-  const cells = await screen.findAllByRole("cell");
-  await waitFor(() => expect(benchmarkName).toBeInTheDocument());
-  await waitFor(() => expect(cells).toHaveLength(20));
+  await screen.findByText("pbench_user_benchmark1");
+  const datasetNameOne = screen.queryByText("pbench_user_benchmark1");
+  const datasetNameTwo = screen.queryByText("pbench_user_benchmark2");
+  const datasetNameThree = screen.queryByText("pbench_user_benchmark3");
+  const datasetNameFour = screen.queryByText("pbench_user_benchmark4");
+  const datasetNameFive = screen.queryByText("pbench_user_benchmark5");
+  expect(datasetNameOne).toBeInTheDocument();
+  expect(datasetNameTwo).toBeInTheDocument();
+  expect(datasetNameThree).toBeInTheDocument();
+  expect(datasetNameFour).toBeInTheDocument();
+  expect(datasetNameFive).toBeInTheDocument();
 });
 
 test("row is favorited after clicking on favorite icon", async () => {
   render(<AppWrapper />);
-  await screen.findByText("dhcp1");
+  await screen.findByText("pbench_user_benchmark1");
   const starBtn = screen.getAllByRole("button", {
-    name: /not starred/i,
+    name: 'Not starred',
   });
   fireEvent.click(starBtn[0]);
   fireEvent.click(starBtn[1]);
   const favoriteBtn = screen.getByRole("button", {
-    name: /see favorites button/i,
+    name: 'see favorites button',
   });
   fireEvent.click(favoriteBtn);
-  const favoriteCell = screen.getAllByRole("cell");
-  expect(favoriteCell).toHaveLength(8);
+  const datasetNameOne = screen.queryByText("pbench_user_benchmark1");
+  const datasetNameTwo = screen.queryByText("pbench_user_benchmark2");
+  const datasetNameThree = screen.queryByText("pbench_user_benchmark3");
+  const datasetNameFour = screen.queryByText("pbench_user_benchmark4");
+  const datasetNameFive = screen.queryByText("pbench_user_benchmark5");
+  expect(datasetNameOne).toBeInTheDocument();
+  expect(datasetNameTwo).toBeInTheDocument();
+  expect(datasetNameThree).not.toBeInTheDocument();
+  expect(datasetNameFour).not.toBeInTheDocument();
+  expect(datasetNameFive).not.toBeInTheDocument();
+});
+test("data is filtered based on value in search box", async () => {
+  render(<AppWrapper />);
+  await screen.findByText("pbench_user_benchmark1");
+  const searchBox = screen.getByPlaceholderText('Search Dataset');
+  fireEvent.change(searchBox, { target: { value: "pbench_user_benchmark2" } });
```

Comment:
Just as a general thought (that is, I'm not necessarily requesting that you make a change), since the first element that the test wants to interact with is `searchBox`, it would make better sense for the `await` statement at line 67 to wait for _that_ element to appear.  This would also make the test more resilient to changes in the mock dataset.

Reply:
The rendering of `searchBox` on the  web page is more of a synchronous task with no practical waiting time involved.So using await for that does not make much sense.But the data fetched from the API is asynchronous,so we have to wait for that data to appear completely before we interact with it, hence use of await